src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -200,6 +200,29 @@ internal void SetEntry(CacheEntry entry)
 
                 priorEntry?.InvokeEvictionCallbacks();
             }
+            else if (_options.HasSizeLimit && priorEntry != null && priorEntry.Size >= entry.Size)
```

Comment:
Would it be possible to avoid a separate case here by passing priorEntry to `UpdateCacheSizeExceedsCapacity`?  @Tratcher [mentioned in the issue](https://github.com/dotnet/runtime/issues/36039#issuecomment-1209969507) that he thought the cause here was that that method didn't consider the size of the evicted item.

Reply:
I agree, for this case we should run the same logic we run for not exceeding capacity.

Reply:
https://github.com/dotnet/runtime/pull/103931